mlrun/model.py

```diff
@@ -774,6 +774,22 @@ def validate_notification_params(self):
 
         notification_class.validate_params(secret_params | params)
 
+    def fill_params_from_secret_params(self):
+        """
+        Fill the notification params from the secret_params.
+        """
+        if "secret" in self.secret_params:
+            secret = self.secret_params["secret"]
+            secret_value = mlrun.get_secret_or_env(secret)
+            if secret_value:
+                try:
+                    self.params.update(json.loads(secret_value))
```

Comment:
this worries me. you are taking a secret and saving it in plaintext params. what if we then save the notification? the secret will not be masked. I'm looking at the `SlackNotification` class and I see it gets the secret as you did here but it doesn't save it to params. Perhaps the change should be there?

Reply:
[Here](https://github.com/mlrun/mlrun/blob/74cc3e1bd4a2fb008945de5c8aa6ee51d34bf54e/mlrun/utils/notifications/notification_pusher.py#L221) we also update params from secret_params, so should we also change this behavior?

Reply:
In general, I wouldn't do this change in the `SlackNotification` class since it's just one type of notification and we need to support secrets for all of the types.
In general the classes that inherit from `NotificationBase` do not have `secret_params` at all - they only have `params` and they are not meant to be serialized, but rather just used for sending the notifications. I believe they are only used in the `NotificationPusher`. 
So, maybe this part should be done when they (the `NotificationBase` classes) are created - I'm not sure exactly where that happens, I guess @quaark or @Yacouby can indeed help there.